internal/k8s/objectmeta.go

```diff
@@ -0,0 +1,37 @@
+// Copyright © 2020 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Object is any Kubernetes object that has an ObjectMeta
+type Object interface {
+	metav1.ObjectMetaAccessor
+}
+
+// Meta holds the name and namespace of a Kubernetes object.
+type Meta struct {
+	Name, Namespace string
+}
+
+// ToMeta returns the Meta of an Object
+func ToMeta(obj Object) Meta {
+	m := obj.GetObjectMeta()
+	return Meta{
+		Name:      m.GetName(),
+		Namespace: m.GetNamespace(),
+	}
+}
```

Comment:
I can't find a place where this copy of the `Meta` type or `ToMeta` is used. Can we remove this?

Reply:
It's used in tests a lot, I think.

Reply:
I don't see it:
```
$ git grep k8s.ToMeta
$ git grep ToMeta
internal/k8s/objectmeta.go:// ToMeta returns the Meta of an Object
internal/k8s/objectmeta.go:func ToMeta(obj Object) Meta {
```

There is a `toMeta` in internal/dag/cache.go, is that it?

Reply:
ah, for posterity - I had moved this to `k8s` in an earlier draft of this PR, but had missed it when starting from scratch. I've renamed this type to `FullName` and moved it to `k8s`.